### PR TITLE
LibWeb: Refactor WebAssembly API to use spec-defined promise AOs

### PR DESCRIPTION
**Pull request by ADKaster**

### CODE REVIEW

File: Tests/LibWeb/Text/expected/Wasm/WebAssembly-instantiate.txt [hunk 1]
```diff
@@ -1,16 +1,52 @@
 -------------
 ArrayBuffer
 -------------
-Hello from wasm!!!!!!
```
Comment (ADKaster) [inline Tests/LibWeb/Text/expected/Wasm/WebAssembly-instantiate.txt:4]: I don't understand why this worked before, if we have always failed to do stuff with the imports?

Comment (ADKaster) [reply Tests/LibWeb/Text/expected/Wasm/WebAssembly-instantiate.txt:4]: #fixed

**Comment by ADKaster** [comment]
Euhh. This ASAN error looks pretty bad. Looks like module instantiation created and then freed some memory that is later referenced by JS? https://github.com/LadybirdBrowser/ladybird/actions/runs/11534938625/job/32109500454?pr=1982#step:17:1149

**Comment by ADKaster** [comment]
Worked around the ASAN bug and filed an issue. It's prettttty bad. <https://github.com/LadybirdBrowser/ladybird/issues/1986>

**Comment by nico** [comment]
follow-up compile fix: https://github.com/LadybirdBrowser/ladybird/commit/72f5fac2ffb48f1caab332baa3e0d1ed2581df99

**Pull request merged**